specs/lib/updateVersionSpec.js

```diff
@@ -0,0 +1,265 @@
+                        technique: 'technique',
+                        values: {
+                            ambient: [0.0, 0.0, 0.0, 1.0]
+                        }
+                    }
+                }
+            },
+            meshes: {
+                mesh: {
+                    primitives: [
+                        {
+                            primitive: WebGLConstants.TRIANGLES
+                        }
+                    ]
+                }
+            },
+            nodes: {
+                node: {
+                    rotation: [0.0, 0.0, 1.0, 0.0],
+                    instanceSkin: {
+                        skeletons: ['skeleton'],
+                        skin: 'skin',
+                        meshes: ['mesh']
+                    }
+                }
+            },
+            techniques: {
+                technique: {
+                    pass: 'pass',
+                    passes: {
+                        pass: {
+                            instanceProgram: {
+                                attributes: {
+                                    attribute: 'TEST_ATTRIBUTE'
+                                },
+                                program: 'program',
+                                uniforms: {
+                                    uniform: 'TEST_UNIFORM'
+                                }
+                            },
+                            states: ['TEST_STATE']
+                        }
+                    }
+                }
+            }
+        };
+        updateVersion(gltf, {
+            targetVersion: '1.0'
+        });
+        expect(gltf.asset.version).toBe('1.0');
+        expect(gltf.asset.profile).toEqual({});
+        expect(gltf.version).not.toBeDefined();
+        expect(gltf.extensionsUsed).toEqual(['extension']);
+        var material = gltf.materials.material;
+        expect(material.technique).toBe('technique');
+        expect(material.values).toEqual({
+            'ambient': [0.0, 0.0, 0.0, 1.0]
+        });
+        var primitive = gltf.meshes.mesh.primitives[0];
+        expect(primitive.primitive).not.toBeDefined();
+        expect(primitive.mode).toBe(WebGLConstants.TRIANGLES);
+        var node = gltf.nodes.node;
+        expect(node.rotation).toEqual([0.0, 0.0, 0.0, 1.0]);
+        expect(node.skeletons).toEqual(['skeleton']);
+        expect(node.skin).toBe('skin');
+        expect(node.meshes).toEqual(['mesh']);
+        var technique = gltf.techniques.technique;
+        expect(technique.pass).not.toBeDefined();
+        expect(technique.passes).not.toBeDefined();
+        expect(technique.attributes).toEqual({
+            attribute: 'TEST_ATTRIBUTE'
+        });
+        expect(technique.program).toBe('program');
+        expect(technique.uniforms).toEqual({
+            uniform: 'TEST_UNIFORM'
+        });
+        expect(technique.states).toEqual(['TEST_STATE']);
+    });
+
+    it('updates glTF from 1.0 to 1.1', function() {
```

Comment:
Can this also test:
* [ ] `glExtensionsUsed` and `extensionsRequired`
* [ ] `buffer.type` removal
* [ ] `technique.functions.scissor` and `technique.states.enable = 3089` removal
* [ ] Adds all properties that were formerly not required, e.g., `asset`, `buffer.byteLength`, `bufferView.byteLength`, etc.
* [ ] `technique.states.functions.blendColor` / `technique.states.functions.depthRange` parameter changes.

In general, can you look at each bullet in https://github.com/KhronosGroup/glTF/issues/605 and confirm that it is handled?  There are others that I did not mention above.

There are some cases where an exact update won't be possible, and we may have to make a breaking change (to the model itself) and provide a log to the user.  For example, we may have to prepend an underscore to an application-specific parameter semantic or change `camera.perspective.yfov = 0.0` to `camera.perspective.yfov = 1.0`.  In practice, most of these edge cases will not come up.

It is likely cleaner to break this into multiple tests for some of these edge cases.